app/controllers/locks_controller.rb

```diff
@@ -60,11 +60,25 @@ def lock
   def require_project
     case action_name
     when 'create' then
-      @project = Stage.find(params[:lock][:resource_id]).project
+      stage = Stage.find(params[:lock][:resource_id])
```

Comment:
this should only be handles by `access_control.rb` and not copied here ...

Reply:
I agree with this, but it seems that only the project object is passed into access control to make the decision for the locks.  I struggled with my debugger configuration, so maybe I overlooked something... but this is the reason for the hacking singleton method below.  I was basically trying to attach the stage attribute information to the project.  I'll do some more investigating...

Reply:
So I have a question about this:
```ruby 
# app/controllers/locks_controller.rb line 6
 before_action :require_project, if: :for_stage_lock?
```
I was going to originally change this to 
```ruby 
 before_action :require_stage, if: :for_stage_lock?
```
Is there some reason that could not be done?  Are there other use cases that use a `Project` to determine lockability?  It seems the locks apply to 2 things, global, and `Stages`.  Why is it that we pass a `Project` in for the access check?